### PR TITLE
Use apicast 3.0.0 beta3

### PR DESCRIPTION
**Pull request by davidor**
This PR modifies the Dockerfile to use the latest version of Apicast (3.0.0-beta3) as the base image.

It also introduces some minor changes to adapt the code to the new interface of the Apicast method that extract the credentials from a request.

### CODE REVIEW

File: xc/utils.lua [hunk 1]
```diff
@@ -0,0 +1,18 @@
+local _M = { }
+
+-- The Apicast method that extracts the credentials from a request returns a
+-- table that includes the credential values, and also key - values where the
+-- keys represent creentials. So for example, a table that contains both
```
Comment (unleashed) [inline xc/utils.lua]: typo `credentials`

**Comment by andrewdavidmackenzie** [comment]
Want to have something in the readme (or other file) about compatibility with apicast, or what version of apicast is used....and modify it in the PR that changes it..... so people know at a glance to the home page for the repo what it's using...?

**Review by unleashed — APPROVED** [review]
minor typo, fix & merge :+1:

File: xc/utils.lua (hunk 1 above)
Comment (davidor) [reply xc/utils.lua]: Fixed :+1:

**Comment by davidor** [comment]
@andrewdavidmackenzie Yes, we'll include a compatibility section when we release a new version.

**Pull request merged**